src/engine.rs

```diff
@@ -121,17 +121,25 @@ impl Engine {
     }
 
     /// Increases the balance for a given address.
-    pub fn increase_balance(address: &Address, amount: &U256) {
-        let mut balance = Self::get_balance(address);
-        balance += *amount;
-        Self::set_balance(address, &balance);
+    fn increase_balance(address: &Address, amount: &U256) -> Result<(), ExitError> {
+        let balance = Self::get_balance(address);
+        let (new_balance, overflow) = balance.overflowing_add(*amount);
```

Comment:
Thanks for adding the overflow checks! I was going to comment on the original transfer PR, but then saw you had already corrected it here.

I personally [prefer `checked_add`](https://docs.rs/primitive-types/0.9.0/primitive_types/struct.U256.html#method.checked_add) because I think the `Option` type captures our use case a little better (when overflow happens there is no meaningful value), but I also recognize it's mostly a style choice.

Reply:
That works.